test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/contract/HapiEthereumCall.java

```diff
@@ -274,6 +277,8 @@ protected Consumer<TransactionBody.Builder> opBodyDef(HapiApiSpec spec) throws T
                             maxGasAllowance.ifPresent(builder::setMaxGasAllowance);
                         }
                 );
+        spec.registry().saveBytes(ETH_HASH_KEY,
+                ByteString.copyFrom(new Keccak.Digest256().digest(signedEthTxData.encodeTx())));
         return b -> b.setEthereumTransaction(ethOpBody);
```

Comment:
I think you can use `EthTxData.getEthereumHash()` directly. Also this code could be moved on line 271, directly under the initialization of `signedEthTxData`. 

We can also introduce a new variable holding the encoded signed data - `signedEthTxData.encodeTx(),`, because it's used twice in this method.